drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -666,9 +682,10 @@ def _mark_complete_and_publish_grade(self):
 
         # There's no going back from "completed" status to "incomplete"
         self.completed = self.completed or self._is_answer_correct() or not self.attempts_remain
-        grade = self._get_grade()
+        grade = self._calculate_grade()
         # ... and from higher grade to lower
-        if grade > self.grade:
+        current_grade = self._get_grade_if_set()
+        if current_grade is None or grade > current_grade:
```

Comment:
@nasthagiri I believe it was discussed at least once. I can't find any links to prove that, but as far as I can recall not locking UI after correct attempt was intentional - to allow student to experiment with the problem more without the risk of getting lower grade (i.e. to read error item feedback).